### PR TITLE
Add apiclient reducer and reduceReceived

### PR DESCRIPTION
**Pull request by coderkevin**
This adds the base reducer for an apiclient state.
It also adds a recursive reduceReceived function that takes data from the
API and updates a redux state.

To Test:
1. `npm test` and ensure all tests pass.

### CODE REVIEW

File: src/react-redux/reducer/__tests__/apiclient-reducer.spec.js [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+		} );
+
+		it( 'should set error state for a new query', () => {
+			const thingsPage1Action = {
+				type: FRESH_DATA_CLIENT_ERROR,
+				apiName: 'test-api',
+				clientKey: '123',
+				endpointPath: [ 'things' ],
+				params: { pge: 1, perPage: 3 },
+				error: { message: 'Query parameter "pge" invalid.' },
```
Comment (timmyc) [inline src/react-redux/reducer/__tests__/apiclient-reducer.spec.js]: sneaky `a`s

File: src/react-redux/reducer/apiclient-reducer.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+/**
+ * Recursively reduce client endpoint state.
+ * 1. If it goes further down the path, recurse.
+ * 2. Otherwise if this has params, store it in queries.
+ * 3. Otherwise store it on this endpoint state.
+ * @param {Object} [state] Should contain endpoints property.
+ * @param {Object} action { apiName, clientKey, endpointPath, params (optional), data }
+ * @param {path} [path] The remaining path (used for recursion).
```
Comment (timmyc) [inline src/react-redux/reducer/apiclient-reducer.js]: should this be `{String}`?

File: src/react-redux/reducer/apiclient-reducer.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+}
+
+/**
+ * Recursively reduce client error endpoint state.
+ * 1. If it goes further down the path, recurse.
+ * 2. Otherwise if this has params, store it in queries.
+ * 3. Otherwise store it on this endpoint state.
+ * @param {Object} [state] Should contain endpoints property.
+ * @param {Object} action { apiName, clientKey, endpointPath, params (optional), data }
+ * @param {path} [path] The remaining path (used for recursion).
```
Comment (timmyc) [inline src/react-redux/reducer/apiclient-reducer.js]: same comment as above

File: src/react-redux/reducer/apiclient-reducer.js (hunk 2 above)
Comment (coderkevin) [reply src/react-redux/reducer/apiclient-reducer.js]: Oh, nope. But it should be `{Array}`. Good catch!

**Comment by coderkevin** [comment]
Thanks, Timmy!

**Pull request merged**